### PR TITLE
Add build analysis workflow

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Related: https://github.com/thoth-station/user-api/pull/1171

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: build-analysis/base/openshift-templates/build-analysis.yaml [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      This template defines resources needed to run build analysis on OpenShift.
+    template.openshift.io/provider-display-name: "Red Hat, Inc."
+  labels:
+    app: thoth
+    template: build-analysis
+    component: build-analysis
+
+parameters:
+  - name: THOTH_BUILD_ANALYSIS_BASE_IMAGE
+    required: false
```
Comment (harshad16) [inline build-analysis/base/openshift-templates/build-analysis.yaml:20]: please remove the default ones like `required: false` from all the parameters.
as it causes a sync issue with argocd.

File: buildlog-parser/base/argo-workflows/buildlog-parser.yaml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+              insecure: true
+              accessKeySecret:
+                name: argo-artifact-repository-secrets
+                key: accessKey
+              secretKeySecret:
+                name: argo-artifact-repository-secrets
+                key: secretKey
+      container:
+        name: parse-buildlog
+        image: quay.io/thoth-station/buildlog-parser:v0.0.1
```
Comment (harshad16) [inline buildlog-parser/base/argo-workflows/buildlog-parser.yaml:89]: Let's use an imagestream for this so that we can manage it easily.

**Review by harshad16 — COMMENTED** [review]
Lgtm, thanks for adding this
some minor changes requested.

File: build-analysis/base/openshift-templates/build-analysis.yaml (hunk 1 above)
Comment (fridex) [reply build-analysis/base/openshift-templates/build-analysis.yaml]: Thanks, adjusted. :+1:

File: buildlog-parser/overlays/cnv-prod/thoth-notification.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        - name: chat-notification
+          image: registry.access.redhat.com/ubi8/ubi
+          command:
+            - "curl"
+            - "-X"
+            - "POST"
+            - "-H"
+            - "Content-Type: application/json; charset=UTF-8"
+            - "-d"
+            - "{'text':'I have successfully synchronized *buildlog-parser* components to *TEST*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/test-thoth-buildlog-parser|ArgoCD UI> 🚀'}"
```
Comment (goern) [inline buildlog-parser/overlays/cnv-prod/thoth-notification.yaml:22]: this is not test isnt it? and the argocd ui url is not correct

File: buildlog-parser/overlays/cnv-prod/thoth-notification.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        - name: chat-notification
+          image: registry.access.redhat.com/ubi8/ubi
+          command:
+            - "curl"
+            - "-X"
+            - "POST"
+            - "-H"
+            - "Content-Type: application/json; charset=UTF-8"
+            - "-d"
+            - "{'text':'🔥 *FAILED* syncing *buildlog-parser* components to *TEST*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/test-thoth-buildlog-parser|ArgoCD UI>'}"
```
Comment (goern) [inline buildlog-parser/overlays/cnv-prod/thoth-notification.yaml:53]: same thing

File: buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        - name: chat-notification
+          image: registry.access.redhat.com/ubi8/ubi
+          command:
+            - "curl"
+            - "-X"
+            - "POST"
+            - "-H"
+            - "Content-Type: application/json; charset=UTF-8"
+            - "-d"
+            - "{'text':'I have successfully synchronized *buildlog-parser* components to *TEST*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/test-thoth-buildlog-parser|ArgoCD UI> 🚀'}"
```
Comment (goern) [inline buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml:22]: this is not TEST

File: buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        - name: chat-notification
+          image: registry.access.redhat.com/ubi8/ubi
+          command:
+            - "curl"
+            - "-X"
+            - "POST"
+            - "-H"
+            - "Content-Type: application/json; charset=UTF-8"
+            - "-d"
+            - "{'text':'🔥 *FAILED* syncing *buildlog-parser* components to *TEST*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/test-thoth-buildlog-parser|ArgoCD UI>'}"
```
Comment (goern) [inline buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml:53]: not TEST

**Review by goern — CHANGES_REQUESTED** [review]

File: buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml (hunk 6 above)
Comment (fridex) [reply buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml]: Thanks, adjusted in b14bfd5.

File: buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml (hunk 5 above)
Comment (fridex) [reply buildlog-parser/overlays/ocp4-stage/thoth-notification.yaml]: Thanks, adjusted in b14bfd5.

File: buildlog-parser/overlays/cnv-prod/thoth-notification.yaml (hunk 4 above)
Comment (fridex) [reply buildlog-parser/overlays/cnv-prod/thoth-notification.yaml]: Thanks, adjusted in b14bfd5.

File: buildlog-parser/overlays/cnv-prod/thoth-notification.yaml (hunk 3 above)
Comment (fridex) [reply buildlog-parser/overlays/cnv-prod/thoth-notification.yaml]: Thanks, adjusted in b14bfd5.

**Review by goern — APPROVED** [review]

**Pull request merged**